proposed/dotnet-sdk-unified-config.md

```diff
@@ -0,0 +1,150 @@
+# A unified configuration file and experience for the .NET SDK
+
+**Owner** [Chet Husk](https://github.com/baronfel) | [Daniel Plaisted](https://github.com/dsplaisted)
+
+The .NET SDK is a product that consists of a number of different tools, all owned by different teams, all operating at slightly different levels of interaction. Some tools operate on the entire repository, some tools on a single project, and yet other tools on a single file.
+
+Because of the disparate nature of these tools, the configuration for each tool is different. This makes it difficult for users to understand how to configure the tools, and difficult for the tools to share configuration information. This document proposes a single, centralized document for repository-wide configuration that tools belonging to the SDK (as well as SDK-adjacent 1st- and 3rd-party tools) can use to share configuration information in a way that has
+
+* a consistent user experience
+* a well supported IDE experience
+* is shareable across team members and trackable in source control
+
+## What exists today
+
+Today there are several places where teams have put configuration for repo- or project-level configuration for tooling:
+
+* global.json files
+* NuGet.config
+* .config/dotnet-tools.json
+* .runsettings (dotnet test)
+
+In addition, several tools are only configurable via environment variables, which are not easily shareable or trackable in source control (due to the lack) of support for .env or other such conventions in the `dotnet` CLI.
```

Comment:
Glad to see [`.env`](https://hexdocs.pm/dotenvy/dotenv-file-format.html) mentioned here. It may as well be a candidate if we are looking for something other than JSON or XML because of its simplicity and its ability to expand values dynamically from environment variables. Python, Docker Compose, Node.js (`node --env-file` new in v20) etc. ecosystems are using it with great success.

Reply:
FWIW, corerun (the internal host for coreclr testing) also uses `.env` since https://github.com/dotnet/runtime/commit/f3c705ef291ff89b53220a31d8321355471d1937, so using it right inside corehost won't be a problem either (if it comes to that).